### ISSUE TITLE
Output type name of MyCLabsEnum includes full namespace

### ISSUE THREAD

**Issue by oojacoboo** [thecodingmachine/graphqlite#232]
When you're outputting an Enum, the type is represented as the full `MyCLabsEnum___App__Enum__TaskStatus` namespace with the \ replaced with _.

I looked into the MyCLabsEnumTypeMapper and see where this is happening here, but I'm not really sure on this logic and seems like it'd require a good deal of digging before I could figure it out. I tried replacing the name in the EnumType object that's created to (new \ReflectionClass($enumClass))->getShortName() but webonyx complains that the type must be a unique name. So, there is clearly more to it here.

**Comment by moufmouf** [comment]
`MyCLabsEnum___App__Enum__TaskStatus` is indeed a very complex name.
I thought I needed to be able to have a bijection between the name and the type (most notably, I thought I needed to be able to find the enum type from the input name). It turns out I'm wrong. Webonyx does not complain (too much) if I don't map the enum name to the enum type.

So we can try to change that.

I guess a good name would be the name of the class. So the `App\Enum\TaskStatus` class would be automatically mapped to an enum named `TaskStatus`.

My problem with this is that we have absolutely no way to handle the case where 2 enum classes have the same name but different namespaces.
We would need a way to override the enum type name.

Maybe a `@EnumType(name="SomeOtherName")` annotation?

My second issue is that strictly speaking, changing the name of the enum types is a breaking change.

I released GraphQLite 4 not too long ago and #227 shows that enums are probably not used widely yet. So a breaking change here even in a minor release might be acceptable. What do you think?

**Comment by oojacoboo** [comment]
@moufmouf I think the fix needs to get out ASAP.  I'd agree it's probably lightly used.  Also, I'm not sure how much of a break that'd be.  How many clients are relying on the type name of Enums?  I'm sure that's a thing, but less so than other design breaks.

As for the annotation, I'm all for keeping the annotation library tighter, so my vote would be for reusing the `@Type` annotation.

**Comment by moufmouf** [comment]
Thanks for the feedback @oojacoboo .

> I think the fix needs to get out ASAP. I'd agree it's probably lightly used. Also, I'm not sure how much of a break that'd be.

I'll introduce the fix in v4.0.2 and document the BC break.

> As for the annotation, I'm all for keeping the annotation library tighter, so my vote would be for reusing the @Type annotation.

I tried to go that way, but it is actually quite complex as the @Type annotation has a semantical meaning that is very different. @Type is expected to be reserved for output types and when I'm scanning output types, I don't want to introduce extra "if" conditions to see if I extend from `Enum` or not (that would make the code less generic and harder to maintain). I introduced another annotation for this, sorry!
